mautic/mautic
# Batch email change category

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴 
| New feature/enhancement? (use the a.x branch)      | 🟢
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🟢 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #... 

## Description

This PR addresses the feature discussed [here](https://forum.mautic.org/t/batch-email-category-update/33106/6):

To allow changing the email category for multiple messages at a time.

The UI looks like this:

![image](https://github.com/user-attachments/assets/feb0cdd6-8ffd-492c-8fa8-e807c59054c0)
![image](https://github.com/user-attachments/assets/8e8115bf-eacd-48fb-857d-5db4715ca051)

---
### 📋 Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Login
3. Go to /s/emails
4. If you don't have any email written create a few and put them in different categories
5. From the email list, select some
6. Click on the batch actions menu
7. Select *Change Category*
8. Select a category
9. Save
10. Refresh the page
11. Every previously selected email should belong to the selected category
